js/main.js

```diff
@@ -0,0 +1,177 @@
+      extras: 'url_sq'
+    },
+    success: function(data) {
+      if (data.photos.photo.length) {
+        var src = data.photos.photo[0].url_sq;
+        var contentString = '<img src="' + src + '" alt="">';
+        var infowindow = new google.maps.InfoWindow({
+            content: contentString
+        });
+
+        var marker = new google.maps.Marker({
+            position: new google.maps.LatLng(latitude, longitude),
+            map: map,
+        });
+        google.maps.event.addListener(marker, 'click', function() {
+          infowindow.open(map, marker);
+        });
+      }
+    }
+  });
+};
+
+/**
+ * HeatMap constructor
+ *
+ * @constructor
+ * @param {number} latitude The latitude value for the center of the map
+ * @param {number} longitude The longitude value for the center of the map
+ */
+var HeatMap = function(startingLatitude, startingLongitude) {
+  this.startingLatitude = startingLatitude;
+  this.startingLongitude = startingLongitude;
+  this.map = new google.maps.Map(document.getElementById('map-canvas'), {
+      zoom: 3,
+      center: new google.maps.LatLng(this.startingLatitude, this.startingLongitude),
+      mapTypeId: google.maps.MapTypeId.SATELLITE
+  });
+  this.mapPoints = [];
+  this.latLngObjArray = [];
+};
+
+/**
+ * Get the coordinates to plot on the map
+ *
+ * @param {array} The list of places returned by the api
+ */
+HeatMap.prototype.getMapPoints = function(places) {
+  var latitude = '';
+  var longitude = '';
+
+  for (var i = 0, count = places.length; i < count; i++) {
+    latitude = places[i].latitude;
+    longitude = places[i].longitude;
+    this.mapPoints.push({latitude: latitude, longitude: longitude});
+    this.formatMapPoints(latitude, longitude);
+  }
+};
+
+/**
+ * Format the points to be used by google api
+ *
+ * @param {number} latitude The latitude of the point to format
+ * @param {number} longitude The longitude of the point to format
+ */
+HeatMap.prototype.formatMapPoints = function(latitude, longitude) {
+  var latLngObj = new google.maps.LatLng(latitude, longitude);
+  this.latLngObjArray.push(latLngObj);
+};
+
+/**
+ * Add heat layer to map
+ *
+ * @param {object} tag The the tag to be mapped
+ * @param {boolean} showPhotos If true, shows top photo for each location
+ */
+HeatMap.prototype.addHeatLayer = function(tag, showPhotos) {
+  this.draw = function(places) {
+    this.getMapPoints(places);
+
+    var heatmap = new google.maps.visualization.HeatmapLayer({
+      data: this.latLngObjArray,
+      radius: 30
+    });
+    heatmap.setMap(this.map);
+    if (showPhotos) {
+      this.addPhotoLayer(tag);
+    }
+  };
+  tag.getPlaces(this.draw.bind(this));
+};
+
+/**
+ * Add photos to map
+ *
+ * @param {object} tag The the tag to be mapped
+ */
+HeatMap.prototype.addPhotoLayer = function(tag) {
+  var photoCount = this.mapPoints.length;
+  var latitude = 0;
+  var longitude = 0;
+
+  $('.message').addClass('show').find('.tag_name').text(tag.searchTerm);
```

Comment:
Even if you can chain jQuery calls, I generally don't, especially if they are doing different things.